python/cudnn/api_base.py

```
@@ -647,20 +684,19 @@ def _is_f16(self, tensor_or_dtype: torch.Tensor | torch.dtype | TensorDesc) -> b
         """
         if tensor_or_dtype is None:
             return False
-        dtype = tensor_or_dtype.dtype if isinstance(tensor_or_dtype, (torch.Tensor, TensorDesc)) else tensor_or_dtype
-        return dtype in {torch.float16, torch.bfloat16}
+        dtype = tensor_or_dtype.dtype if isinstance(tensor_or_dtype, TensorDesc) or _is_framework_tensor(tensor_or_dtype) else tensor_or_dtype
+        return _convert_to_cutlass_data_type_or_none(dtype) in {cutlass.Float16, cutlass.BFloat16}
 
     def _get_innermost_stride_dim(self, tensor: torch.Tensor, name: str = "") -> int:
```

Ruff (0.16.1)

[error] Undefined name `torch`

(F821)